Fix `chia farm summary` aborting early if no local full node present

xearl4 opened the pull request:
Before 2.0, even if no local full node was present, `chia farm summary` would still show useful information:

	$ chia farm summary
	Farming status: Not available
	Local Harvester
	   1 plots of size: 101.320 GiB on-disk, 101.400 GiBe (effective)
	Plot count for all harvesters: 1
	Total size of plots: 101.320 GiB, 101.400 GiBe (effective)
	Estimated network space: Unknown
	Expected time to win: Unknown
	For details on farmed rewards and fees you should run 'chia start wallet' and 'chia wallet show'

However, since 2.0, that is no longer the case. `chia farm summary` simply aborts with an exception:

	$ chia farm summary
	Error: Connection error: ClientConnectorError: Cannot connect to host localhost:8555 ssl:<ssl.SSLContext object at 0x7fe0edd533c0> [Connect call failed ('127.0.0.1', 8555)]
	Check if full node rpc is running at 8555
	This is normal if full node is still starting up

This fixes this particular regression which is rather annoying for farmers farming to a remote node.

Fixes #16164. Related to issue #9615, but that issue is wider in scope.

xearl4 committed:
Fix `chia farm summary` aborting early if no local full node present

Before 2.0, even if no local full node was present, `chia farm summary`
would still show useful information:

	$ chia farm summary
	Farming status: Not available
	Local Harvester
	   1 plots of size: 101.320 GiB on-disk, 101.400 GiBe (effective)
	Plot count for all harvesters: 1
	Total size of plots: 101.320 GiB, 101.400 GiBe (effective)
	Estimated network space: Unknown
	Expected time to win: Unknown
	For details on farmed rewards and fees you should run 'chia start wallet' and 'chia wallet show'

However, since 2.0, that is no longer the case. `chia farm summary`
simply aborts with an exception:

	$ chia farm summary
	Error: Connection error: ClientConnectorError: Cannot connect to host localhost:8555 ssl:<ssl.SSLContext object at 0x7fe0edd533c0> [Connect call failed ('127.0.0.1', 8555)]
	Check if full node rpc is running at 8555
	This is normal if full node is still starting up

This fixes this particular regression which is rather annoying for
farmers farming to a remote node.

Fixes #16164. Related to issue #9615, but that issue is wider in scope.

altendky reviewed (changes requested):
It would be nice to provide at least some information about the failure, in addition to the other information.  At least for most errors.  Maybe we could specifically handle `ClientConnectorError` in a quiet way and provide noisier output for all other exceptions?  I'm ok with continuing on in both cases and providing the rest of the information we can provide.

altendky commented:
Also...  Thank you!

xchdata1 commented:
Yes, that's a good idea. So I'll silently handle `ClientConnectorError` and print the exception for all other exceptions.

altendky commented on chia/cmds/farm_funcs.py:
```diff
@@ -89,7 +90,14 @@ async def summary(
+        blockchain_state = await get_blockchain_state(rpc_port, root_path)
+    except CliRpcConnectionError:
+        pass
+    except Exception as ex:
+        print(ex)
```
Converting an exception to a string gives only the text, not the type or the traceback.  In cases like `some_dict[a_not_present_key]` this gives you just a string of the key.  Some exceptions don't even bother having messages at all and would just be blank.

```suggestion
    except Exception as e:
        print(traceback.print_exc(file=sys.stderr))
```

I think there could be some refinement around this, including applying similar handling to other such requests in this function, but having the exception at the top of the output and on stderr still leaves the rest of the output intact, including on redirect to files or piping to other programs etc.

You may have to add imports for `sys` and `traceback`.

Thanks again.

earl replied:
My bad, got the exceptions mixed up. The connection error is wrapped in a `CliRpcConnectionError` which always provides a nice message. But this is of course precisely _not_ handling the `CliRpcConnectionError` :)

Fixed. And I also adjusted the exception handling for the similar local wallet connection.

xchdata1 replied:
Thanks! Got the exceptions mixed up. The connection error is wrapped in a CliRpcConnectionError which always provides a nice message. But this is of course precisely not handling the CliRpcConnectionError :)

Fixed. And I also adjusted the exception handling for the similar local wallet connection.

xearl4 committed:
Improve exception reporting for `chia farm summary`

With this change, only the exceptions when failing to connect to a local
full node and/or wallet are suppressed, as these cases are already
properly handled by adjusted command output. Any other exception is
printed to stderr.

altendky commented on chia/cmds/farm_funcs.py:
```diff
@@ -1,5 +1,8 @@
 from __future__ import annotations
 
+import sys
+import traceback
+
```
https://github.com/Chia-Network/chia-blockchain/actions/runs/6274090643/job/17038962616?pr=16387#step:9:136
```suggestion
import sys
import traceback
```

isort is being picky here.  I will also try to make the CI output for this more descriptive.  I ran it locally to confirm this was the change it wants.

altendky replied:
And...  github doesn't even show the removed blank line...

xchdata1 replied:
Yes, complains locally as well. Fixed.

altendky reviewed (dismissed):
Looks good, pending CI.  Thanks for the work and the contribution.

altendky commented:
Actually, sorry, could you rebase and retarget this to `release/2.1.0`?  Or I can do that, if you've allowed me access to this branch via GitHub.  Either way.

xchdata1 commented:
> Actually, sorry, could you rebase and retarget this to `release/2.1.0`?

Sure, done. And thanks for your review!
